src/main/java/com/powsybl/openloadflow/ac/outerloop/IncrementalTransformerVoltageControlOuterLoop.java

```diff
@@ -56,6 +57,24 @@ public String getName() {
         return NAME;
     }
 
+    public static List<LfBus> getControlledBuses(LfNetwork network) {
+        return network.getBuses().stream()
+                .filter(bus -> bus.isVoltageControlled(VoltageControl.Type.TRANSFORMER))
+                .filter(bus -> bus.getVoltageControl(VoltageControl.Type.TRANSFORMER).orElseThrow().getMergeStatus() == VoltageControl.MergeStatus.MAIN)
+                .filter(bus -> !bus.getVoltageControl(VoltageControl.Type.TRANSFORMER).orElseThrow().isHidden(true))
+                .collect(Collectors.toList());
+    }
+
+    public static List<LfBranch> getControllerElements(LfNetwork network) {
+        return network.getBuses().stream()
+                .filter(bus -> bus.isVoltageControlled(VoltageControl.Type.TRANSFORMER))
+                .filter(bus -> bus.getVoltageControl(VoltageControl.Type.TRANSFORMER).orElseThrow().getMergeStatus() == VoltageControl.MergeStatus.MAIN)
+                .filter(bus -> !bus.getVoltageControl(VoltageControl.Type.TRANSFORMER).orElseThrow().isHidden(true))
+                .flatMap(bus -> bus.getTransformerVoltageControl().orElseThrow().getMergedControllerElements().stream())
+                .filter(Predicate.not(LfBranch::isDisabled))
+                .collect(Collectors.toList());
+    }
+
```

Comment:
You have forgotten to change the list of controller elements to build the context.